source/test/test.ts

```diff
@@ -58,20 +58,27 @@ test('fail if `typings` property is used instead of `types`', async t => {
 });
 
 test('fail if tests don\'t pass in strict mode', async t => {
-	const diagnostics = await m({
-		cwd: path.join(__dirname, 'fixtures/failure-strict-null-checks')
-	});
-
-	const {fileName, message, severity, line, column} = diagnostics[0];
-	t.true(/failure-strict-null-checks\/index.test-d.ts$/.test(fileName));
-	t.is(
-		message,
-		`Argument of type 'number | null' is not assignable to parameter of type 'number'.
+	function assert(diagnostics, expectedFilePathRe) {
+		const {fileName, message, severity, line, column} = diagnostics[0];
+		t.true(new RegExp(expectedFilePathRe).test(fileName));
+		t.is(
+			message,
+			`Argument of type 'number | null' is not assignable to parameter of type 'number'.
   Type \'null\' is not assignable to type 'number'.`
+		);
+		t.is(severity, 'error');
+		t.is(line, 4);
+		t.is(column, 19);
+	}
+
+	assert(
```

Comment:
Remove the `assert` function, I'd rather be explicit per test case as currently it's just by accident that the line and column of best test fixtures are the same.

Just split both the `strict-null-checks` and `strict-null-checks-as-default-config-value` in separate tests.